app/uk/gov/hmrc/tradergoodsprofiles/services/RouterService.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.services
+
+import cats.data.EitherT
+import com.google.inject.{ImplementedBy, Inject}
+import play.api.Logging
+import play.api.libs.json._
+import play.api.mvc.Result
+import play.api.mvc.Results.Status
+import uk.gov.hmrc.http.HttpReads.is2xx
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
+import uk.gov.hmrc.tradergoodsprofiles.connectors.RouterConnector
+import uk.gov.hmrc.tradergoodsprofiles.models.GetRecordResponse
+import uk.gov.hmrc.tradergoodsprofiles.models.errors.{RouterError, ServerErrorResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.runtime.universe.{TypeTag, typeOf}
+import scala.util.{Failure, Success, Try}
+
+@ImplementedBy(classOf[RouterServiceImpl])
+trait RouterService {
+
+  def getRecord(eori: String, recordId: String)(implicit hc: HeaderCarrier): EitherT[Future, Result, GetRecordResponse]
+}
+
+class RouterServiceImpl @Inject() (
+  routerConnector: RouterConnector,
+  dateTimeService: DateTimeService
+)(implicit ec: ExecutionContext)
+    extends RouterService
+    with Logging {
+
+  def getRecord(eoriNumber: String, recordId: String)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, Result, GetRecordResponse] =
+    EitherT(
+      routerConnector
+        .get(eoriNumber, recordId)
+        .map {
+          case httpResponse if is2xx(httpResponse.status) => jsonAs[GetRecordResponse](httpResponse.body)
+          case httpResponse                               => Left(handleError(httpResponse.body, httpResponse.status, eoriNumber, recordId))
+        }
+        .recover {
+          case UpstreamErrorResponse(message, status, _, _) => Left(handleError(message, status, eoriNumber, recordId))
```

Comment:
Does this work?! The error response will be caught in the else case above

Reply:
Yes, you are right this is dead code. It will never be triggered. Removing it.